behaviour/graql/explanation/reasoner.feature

```diff
@@ -0,0 +1,502 @@
+#
+# Copyright (C) 2020 Grakn Labs
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+Feature: Graql Reasoning Explanation
+
+  Only scenarios where there is only one possible resolution path can be tested in this way
+
+  Background: Initialise a session and transaction for each scenario
+    Given connection has been opened
+    Given connection delete all keyspaces
+    Given connection open sessions for keyspaces:
+      | test_explanation |
+    Given transaction is initialised
+
+  Scenario: an attribute's existence and ownership can be inferred
+    Given graql define
+      """
+      define
+
+      name sub attribute, value string;
+      company-id sub attribute, value long;
+
+      company sub entity,
+        has name,
+        key company-id;
+
+      company-has-name sub rule,
+      when {
+         $c isa company;
+      }, then {
+         $c has name "the-company";
+      };
+      """
+
+    When graql insert
+      """
+      insert
+      $x isa company, has company-id 0;
+      """
+
+    Then get answers of graql query
+      """
+      match $co has name $n; get;
+      """
+
+    Then concept identifiers are
+      |     | check | value            |
+      | CO  | key   | company-id:0     |
+      | CON | value | name:the-company |
+
+    Then uniquely identify answer concepts
+      | co | n   |
+      | CO | CON |
+
+    Then rules are
+      |                  | when                 | then                            |
+      | company-has-name | { $c isa company; }; | { $c has name "the-company"; }; |
+
+    Then answers contain explanation tree
+      |   | children | vars  | identifiers | rule             | pattern                                                           |
```

Comment:
Ah yes I haven't gone back and revised this since we introduced the new explanation types. Previously it made more sense, and for compactness this field includes the rule name if the explanation is indeed a `RuleExplanation`

Reply:
Have gone with `explanation`